hack/e2e-test.sh

```diff
@@ -53,64 +53,101 @@ make deploy DEPLOY_IMAGE="${HIVE_IMAGE}"
-	echo "************* INSTALL JOB LOG *************"
-	if oc get clusterprovision -l "hive.openshift.io/cluster-deployment-name=${CLUSTER_NAME}" -o jsonpath='{.items[0].spec.installLog}' &> "${ARTIFACT_DIR}/hive_install_console.log"; then
-		cat "${ARTIFACT_DIR}/hive_install_console.log"
-	else
-		cat "${ARTIFACT_DIR}/hive_install_job.log"
+	PLATFORM=${1}
+	CLUSTER_NAME=${BASE_CLUSTER_NAME}-${PLATFORM}
+	LOGS_DIR="${ARTIFACT_DIR}/${PLATFORM}"
+	if mkdir -p "${LOGS_DIR}"
+	then
+		if INSTALL_JOB_NAME=$(oc get job -l "hive.openshift.io/cluster-deployment-name=${CLUSTER_NAME},hive.openshift.io/install=true" -o name) && [ "${INSTALL_JOB_NAME}" ]
+		then
+			oc logs -c hive job/${INSTALL_JOB_NAME} &> "${LOGS_DIR}/hive_install_job.log" || true
+			oc get job ${INSTALL_JOB_NAME} -o yaml &> "${LOGS_DIR}/hive_install_job.yaml" || true
+		fi
+		oc get clusterdeployment -A -o yaml &> "${LOGS_DIR}/hive_clusterdeployment.yaml" || true
+		oc get clusterimageset -o yaml &> "${LOGS_DIR}/hive_clusterimagesets.yaml" || true
+		echo "************* INSTALL JOB LOG *************"
+		if oc get clusterprovision -l "hive.openshift.io/cluster-deployment-name=${CLUSTER_NAME}" -o jsonpath='{.items[0].spec.installLog}' &> "${LOGS_DIR}/hive_install_console.log"; then
+			cat "${LOGS_DIR}/hive_install_console.log"
+		else
+			cat "${LOGS_DIR}/hive_install_job.log"
+		fi
 	fi
 
 	echo ""
 	echo ""
 	echo "Deleting ClusterDeployment ${CLUSTER_NAME}"
-	oc delete --wait=false clusterdeployment ${CLUSTER_NAME}
-	errorOnUninstall=0
-	if ! go run "${SRC_ROOT}/contrib/cmd/waitforjob/main.go" --log-level=debug "${CLUSTER_NAME}" "uninstall"; then
-		errorOnUninstall=1
-	fi
+	oc delete --wait=false clusterdeployment ${CLUSTER_NAME} || :
 
-	if [[ $errorOnUninstall == 1 ]]; then
-		if oc logs job/${CLUSTER_NAME}-uninstall &> "${ARTIFACT_DIR}/hive_uninstall_job.log"; then
+	if ! go run "${SRC_ROOT}/contrib/cmd/waitforjob/main.go" --log-level=debug "${CLUSTER_NAME}" "uninstall"
+	then
+		echo "Waiting for uninstall job failed"
+		if mkdir -p "${LOGS_DIR}" && oc logs job/${CLUSTER_NAME}-uninstall &> "${LOGS_DIR}/hive_uninstall_job.log"
+		then
 			echo "************* UNINSTALL JOB LOG *************"
 			cat "${ARTIFACT_DIR}/hive_uninstall_job.log"
 			echo ""
 			echo ""
-		else
-			echo "Waiting for uninstall job failed"
 		fi
 		exit 1
 	fi
 }
-trap 'teardown' EXIT
 
 echo "Running post-deploy tests"
 make test-e2e-postdeploy
 
-echo "Creating cluster deployment"
-echo "Release image: $RELEASE_IMAGE"
-SRC_ROOT=$(git rev-parse --show-toplevel)
-
-go run "${SRC_ROOT}/contrib/cmd/hiveutil/main.go" create-cluster "${CLUSTER_NAME}" \
-	--ssh-public-key-file="${SSH_PUBLIC_KEY_FILE}" \
-	--pull-secret-file="${PULL_SECRET_FILE}" \
-	--base-domain="${BASE_DOMAIN}" \
-	--release-image="${RELEASE_IMAGE}" \
-	--install-once=true \
-	--uninstall-once=true
+export SRC_ROOT=$(git rev-parse --show-toplevel)
+
+for PLATFORM in "${PLATFORMS[@]}"
```

Comment:
I'm struggling with the thought of one e2e job that covers all platforms. On one hand this is definitely a quicker and easier path to it, and we just install Hive once and then use it to provision all three, which has some benefits in coverage.

It may however be very difficult to get PRs to merge when we need all three to pass and can't re-run specific clouds when they fail.

I'd probably lean towards the approach you've taken though and we can keep an eye on it.